src/OpenTelemetry/Internal/SdkHealthReporter.cs

```diff
@@ -0,0 +1,32 @@
+// <copyright file="SdkHealthReporter.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics.Metrics;
+
+namespace OpenTelemetry.Internal
+{
+    internal class SdkHealthReporter
+    {
+        public SdkHealthReporter(string providerId)
+        {
+            this.BatchExportProcessorDroppedCount = InternalMeter.CreateCounter<long>($"{providerId}.BatchExportProcessorDroppedCount");
```

Comment:
A standard prefix for our health metrics would be useful. Something like `dotnet.sdk.*`. So this metric could be `dotnet.sdk.processor.dropped_count`.

One scenario where this becomes useful is if you're using the collector to process these metrics specially. I can configure my collector to match any metric name that starts with `dotnet.sdk`.

Also I think `provider.id=123` and `processor.type=BatchExportProcessor` would be good as attributes.

Reply:
* If we don't use providerId in the name than this whole thing could be made static and then that just becomes part of the `TagList` on `BatchExportProcessor`?

* Naming it `dotnet.sdk.processor.dropped_count` are we sure about that? The name seems to indicate other processors should also use it? How do we enforce they write the correct tags? And would that prevent us from changing the tags later? Feel like it should be specific a la `dotnet.sdk.batch_processor.dropped_count` 🤷

Reply:
> Feel like it should be specific a la dotnet.sdk.batch_processor.dropped_count

That name sounds fine. My main point is we should be striving towards a convention with a standard prefix. So something along the lines `dotnet.sdk.[sdkcomponent].[somemeasurement]`.

> The name seems to indicate other processors should also use it?

Sure, and maybe this metric is very specific to batch processors, in which case the name you've proposed is fine. Though, I don't want us to lock ourselves into thinking this level of specificity is required for all health metrics we might come up with. Just an example, let's say we have `dotnet.sdk.sampler.total_spans`.

> How do we enforce they write the correct tags?

I'm not convinced the processor should be using the underlying counter directly. I alluded to this on our call, but I was thinking the processor would use a method like.

```csharp
void RecordBatchExportProcessorDroppedCount(long count, string processorName, string signalName)

// Then a processor would call it like
sdkHeathReporter.RecordBatchExporterProcessorDroppedCount(4, "BatchExportProcessor", "Activity")
```

This makes it impossible to not add the expected tags. Also makes it less possible to generate an explosion of tag values.

Reply:
> If we don't use providerId in the name than this whole thing could be made static and then that just becomes part of the TagList on BatchExportProcessor?

would like to discuss this part in SIG today if we get time.

Reply:
@CodeBlanch @alanwest  made some updates. Please have a look when you get chance.